### ISSUE TITLE
Flutter Attaching notification needs a cancel button

### ISSUE THREAD

**Issue by gaaclarke** [Dart-Code/Dart-Code#5372]
**Is your feature request related to a problem? Please describe.**
Yes, https://github.com/Dart-Code/Dart-Code/issues/5371

**Describe the solution you'd like**
There should be a "cancel" button in the notification that the debugger is attempting to attach.

**Describe alternatives you've considered**
n/a

**Additional context**
This is the notification I'm talking about.  In my case, it would never successfully attach, it would be nice to be able to cancel it.  There is a tiny button at the top that can do that, but it doesn't match the design of many other vscode notifications.

<img width="481" alt="Screenshot 2024-12-16 at 4 02 46 PM" src="https://github.com/user-attachments/assets/485dc7ba-fd66-4ab2-a8c1-d59df6718ec2" />

**Comment by DanTup** [comment]
We don't support cancellation on the notification because this is just a progress indicator triggered by the debug adapter. It just sends some text and we show it as the current progress. There's no concept of cancelling progress.

Instead, if you don't want this debug session, you'd have to click the Stop button on the debug toolbar, which will send a `terminate` request to the debug adapter and it should stop and tear down.

If that doesn't work (or leaves processes, or anything), that's a bug.

**Comment by gaaclarke** [comment]
I don't agree your take on this, let me try to convince you.  So, remember I'm in an error state where the attaching dialog never resolves, I need to cancel to progress.  The notification pops up after I've chosen to attach and it is animating, there is an indeterminate progress indicator drawing your attention to it.  It's over 10x bigger than the debugging bar that appears at the same time and the debugging bar is not animated.

The dialog reads "Attaching...", I think a button on there that says "Cancel" would clearly mean to users that you "cancel attaching".  I don't think that would be confusing.
